pkg/sethandler/controller.go

```diff
@@ -43,9 +43,9 @@ type checkpointFile struct {
 
 //SetHandler is the data set encapsulating the configuration data needed for the CPUSetter Controller to be able to adjust cpusets
 type SetHandler struct {
-	poolConfig types.PoolConfig
-	cpusetRoot string
-	k8sClient  kubernetes.Interface
+	PoolConfig types.PoolConfig
```

Comment:
I guess these fields were made public for the unit tests. Would it be possible to arrange tests so that these could be private ?

Reply:
that's coming from me
if we make them private, it means the test needs to be part of the same package. but then it has access to all the functions to test directly.
some unit testing best practice I have used within my R&D in the years is to only test packages through their public interfaces, or interfaces through which the control loops are invoked.
If we can test every little peace in the middle on its own, it is very easy to test things which might not be even invoked during a normal execution.
Also, testing private interfaces make the tests more brittle, because internal function signatures tend to change a lot, while publics or entrypoints are more or less constant

yes, the prices of this approach is making some stuff public :) we could define getters on this struct, but IMO it would be just unnecessarily clutter. this makes the data struct also reusable in other projects, you never know, someone might find it useful :)

Reply:
That is valid point but we can also have two levels of testing:
- Tests that test package through public API
- Tests that tests packages internals.

Sometimes we are not able to (easily) cover all functions by testing through public API so we need some tests to internals. Maybe that is not the case here as the coverage seems to be quite high.

Exposing the internal fields is anyway problematic. If someone starts to use this package and uses the fields directly, it makes it really hard to do refactoring (field renaming etc.) without affecting the users of this package.

Reply:
we always had the argument that if you can't reach a code part through the public API, you don't need it :) yeah, some error scenarios are harder to test, but otherwise 90% coverage usually can be easily achieved with this approach too

okay, understand your reasoning regarding exposing members. let's add getters then for those we need, and keep the members private.
this still enables us to do this kind of testing without exposing the internals

Reply:
@tamas-laczko so, what's up with this comment? IMO this is the only one still open

Reply:
fixed, also modified temp fs removal.